core/internal/notifier/coordinator.go

```diff
@@ -326,6 +326,11 @@ func (nc *Coordinator) manageEvalLoop() {
 		for !nc.App.ZookeeperConnected {
 			time.Sleep(100 * time.Millisecond)
 		}
+
+		err := lock.Unlock()
```

Comment:
Since `err` is defined above, this should fail to build. Using assignment instead of short variable declaration should work, though.

```go
err = lock.Unlock()
```

This is interesting that it still hasn't received any attention upstream. The tests to hit this should (or do?) live in `coordinator_race_test.go`. I have some experience with Go, I'll take a look and see if anything sticks out to me there.

Reply:
At least someone around here knows go! Thanks a ton for this! I haven’t had a chance to build it yet, so that’d explain it. If you know more about testing go and everything, I’d welcome some help in getting a pipeline up for this Monday if you have some time.

Reply:
I'd be happy to take a look at it with you. I have a couple of questions about this method (eg. whether this lock is expected to be ephemeral and auto-unlocked on disconnection from zookeeper) but I think this change will probably work for our use case anyway.

The tests seem to mostly be around a mocked server, so we can fix up the tests to handle the new behavior in a sane way, but I'm not sure it will help us understand the expected behavior.